### PR TITLE
ARXIVNG-3039: Dockerfile and helm chart for classic API reimplementation

### PR DESCRIPTION
**Pull request by mhl10**
This PR adds a Dockerfile and helm chart for the classic API.

### CODE REVIEW

**Comment by bmaltzan** [comment]
Seems like an external issue building it right now. (I can use the previous local version of search 0.5.5 to run it.) 

```
docker-compose build
Step 1/20 : FROM arxiv/search:0.5.6
ERROR: Service 'agent' failed to build: manifest for arxiv/search:0.5.6 not found: manifest unknown: manifest unknown
```

```
docker build -f ./Dockerfile .
Could not retrieve mirrorlist https://mirrors.iuscommunity.org/mirrorlist?repo=ius-centos7&arch=x86_64&protocol=http error was
14: curl#6 - "Could not resolve host: mirrors.iuscommunity.org; Name or service not known"
```

**Comment by mhl10** [comment]
> Seems like an external issue building it right now. (I can use the previous local version of search 0.5.5 to run it.)
> 
> ```
> docker-compose build
> Step 1/20 : FROM arxiv/search:0.5.6
> ERROR: Service 'agent' failed to build: manifest for arxiv/search:0.5.6 not found: manifest unknown: manifest unknown
> ```
> 
> ```
> docker build -f ./Dockerfile .
> Could not retrieve mirrorlist https://mirrors.iuscommunity.org/mirrorlist?repo=ius-centos7&arch=x86_64&protocol=http error was
> 14: curl#6 - "Could not resolve host: mirrors.iuscommunity.org; Name or service not known"
> ```

Oops, that should be `docker build -f ./Dockerfile-classic-api`.

**Comment by mhl10** [comment]
> Oops, that should be `docker build -f ./Dockerfile-classic-api`.

Since there's no Docker-compose file for the classic API just yet, building `Dockerfile-classic-api` on its own may not be that interesting.

File: Dockerfile-classic-api [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+RUN pipenv install
+
+ENV PATH "/opt/arxiv:${PATH}"
+
+ADD schema /opt/arxiv/schema
+ADD mappings /opt/arxiv/mappings
+ADD search /opt/arxiv/search
+ADD wsgi-classic-api.py config/uwsgi-classic-api.ini /opt/arxiv/
+
+ENV LC_ALL en_US.utf8
```
Comment (bdc34) [inline Dockerfile-classic-api]: Since each ENV makes a layer, move these to the start of the Dockerfile.

File: Dockerfile-classic-api [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+FROM arxiv/base:0.16.6
+
+WORKDIR /opt/arxiv
+
+# Install MySQL.
+# RUN yum install -y which mysql mysql-devel
+
+# Add Python application and configuration.
+ADD classic-api.py /opt/arxiv/
```
Comment (bdc34) [inline Dockerfile-classic-api]: Move this to after the pipenv install so the python installed packages can be reused as a layer. 

If the code changes, it invalidates the cache for all following layers.

Comment (mhl10) [reply Dockerfile-classic-api]: please see updated version

File: Dockerfile-classic-api (hunk 1 above)
Comment (mhl10) [reply Dockerfile-classic-api]: please see updated version

**Pull request merged**